File: server-notify-messagerie/NotificationService/Models/MyFriends.cs
using NotificationService.Models;

public class MyFriends
{
    public User User { get; set; }  // User details
    public DateTime CreatedAt { get; set; }  // Date of the friendship
    public int NbMutualFriends{get; set;}
    public List<String>? MutualFriends{get; set;}


    public MyFriends() {}
    public MyFriends(User user, DateTime createdAt, int nbMutualFriends, List<String>? mutualFriends )
    {
        User = user;
        CreatedAt = createdAt;
        MutualFriends = mutualFriends;
        NbMutualFriends = nbMutualFriends;
    }

}

File: client-notify-messagerie/src/interfaces/MyFriends.ts
import { User } from './User'

export interface MyFriends {
    id: string
    user: User;
    mutualFreinds: string[];
    nbMutualFriends: number;
    createdAt: Date;
}

// export interface FriendsResult {
//     totalFriends: number;         
//     friendsList: MyFriends[];    
// }

export interface InvitationsFriends {
    id: string
    user: User;
    nbMutualFriends: number;
    sentAt: Date;
    mutualFreinds: string[];
    status: string;
}

export interface AnswerInvitationRequest {
    userId: string;
    friendId: string;
    answerInvitationChoice: InvitationResponse;
}

export enum InvitationResponse {
    Accepted = 0,
    Rejected = 1,
}

export interface FriendsRequests {
    id: string
    user: User;
    nbMutualFriends: number;
    sentAt: Date;
    mutualFreinds: string[];
    status: string;
}

export interface CancelledFriendRequest {
    userId: string;
    friendId: string;
}
